### PR TITLE
configure: refuse to compile without any database.

### PR DESCRIPTION
**Pull request by rustyrussell**
Fixes: #4054
Changelog-None

### CODE REVIEW

**Comment by cdecker** [comment]
Best error message eva!

ACK https://github.com/ElementsProject/lightning/pull/4063/commits/7725f6c89a39724ee6f17728dfef265ab5ad064d

**Comment by whitslack** [comment]
Is there no way to have `configure` complain about this rather than `make`? (I would have implemented such a check in my PR to make SQLite3 optional, but I didn't see any straightforward way to access the variables emitted by Configurator from within the `configure` script.)

**Comment by rustyrussell** [comment]
There is... Let's do that then.

**Comment by cdecker** [comment]
ACK https://github.com/ElementsProject/lightning/pull/4063/commits/b76e158633cbd5abfe980e85059eb0ad060c05b9

File: configure [hunk 1]
```diff
@@ -396,3 +394,16 @@ add_var HAVE_PYTHON3_MAKO "$HAVE_PYTHON3_MAKO"
...
+
+# Now we set them all and check.
+while IFS='=' read VAR VAL; do
+    eval $VAR=\"$VAL\"
+done < $CONFIG_VAR_FILE.$$
+
+if [ "$HAVE_SQLITE3" = 0 -a "$HAVE_POSTGRES" = 0 ]; then
+    # I have no database yet I must schema!)
+    echo "*** We need a database, but neither sqlite3 nor postgres found" >&2
+    exit 1
```
Comment (whitslack) [inline configure:406]: Exiting here would leave the temporary file `$CONFIG_VAR_FILE.$$` in place. Is that what you want?

Comment (rustyrussell) [reply configure:406]: Oops, I assumed there was a "trap" to clean this up (there are other cases we can exit early, due to `set -e`).  Will fix that separately...

**Pull request merged**